server/src/testflinger/api/v1.py

```
@@ -835,6 +836,34 @@ def get_agents_on_queue(queue_name):
     return database.get_agents_on_queue(queue_name)
 
 
+@v1.get("/queues/<queue_name>/jobs")
+def get_jobs_by_queue(queue_name):
+    """Get the jobs in a specified queue along with its state.
+
+    :param queue_name
+        String with the queue name where to perform the query.
+    :return:
+        JSON data with the jobs allocated to the specified queue.
+    """
+    jobs = list(
+        database.mongo.db.jobs.find({"job_data.job_queue": queue_name})
+    )
+
+    if not jobs:
+        return [], HTTPStatus.NO_CONTENT
+
+    jobs_in_queue = [
+        {
+            "job_id": job.get("job_id"),
+            "created_at": job.get("created_at"),
+            "job_state": job.get("result_data", {}).get("job_state"),
```

Using `get` means that if a job doesn't have values for these fields then we simply use `None`. But I am wondering:
- Is there any chance a job won't have a value for these fields?
- If it doesn't, should we actually fail (raise an exception) instead of silently using `None`?

---

I don't think there is, AFAIK once the job is created, it will add the ID, timestamp and set the initial status to "waiting". I don't think there is a reason why any of those values are missing on a job. 

I can add an exception to handle this but ideally, we shouldn't hit any case. I'm guessing that if there is a problem to set any of those values, it will be caught during job creation.

---

This was also covered in #689 and an exception was included in case the server is unable t retrieve any of these keys.